### ISSUE TITLE
jmhJar task is not compatible with Gradle configuration cache

### ISSUE THREAD

**Issue by shoaniki** [melix/jmh-gradle-plugin#229]
**Describe the bug**
The plugin causes an error if the experimental Gradle configuration cache is enabled.

**To Reproduce**
Run `gradle jmh --configuration-cache` in a project that uses this plugin.

The output will include an error along the lines of:
```
FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache problems found in this build.

1 problem was found storing the configuration cache.
- Task `:project:jmhJar` of type `org.gradle.api.tasks.bundling.Jar`: cannot serialize object of type 
'org.gradle.api.internal.project.DefaultProject', a subtype of 'org.gradle.api.Project', as these are not
supported with the configuration cache.
  See https://docs.gradle.org/7.6-rc-3/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
```
This appears to happen for both shadow and non-shadow forms of the jar task.

There's an easy and obvious workaround, of course -- "don't do that" -- but it becomes a chore if the configuration cache is turned on in gradle.properties.

**Comment by asarkar** [comment]
Gradle configuration cache is a rushed, ill-designed and half-baked feature that is incompatible with many of Gradle's own existing APIs. See https://github.com/gradle/gradle/issues/18520

**Comment by melix** [comment]
Can you provide a reproducer? Which version of the plugin are you using? I pushed a [test](https://github.com/melix/jmh-gradle-plugin/commit/783ae4d9501902469f96bca659c081328095ae67) which passes.

**Comment by jjohannes** [comment]
I don't know why the tests do not reproduce this, but all my manual testing indicate that it is a general problem: it is not possible to access `project.zipTree` in this mapping context, which is (presumably) executed close to execution time. Which somehow makes sense, as the actual files are not always available during configuration time yet.

Maybe there is a simpler way to get the `ArchiveOperations` service, but this is the "simplest" fix I could come up with: #261
